custom/enikshay/case_utils.py

```diff
@@ -169,44 +172,52 @@ def get_all_episode_cases_from_person(domain, person_case_id):
     return episode_cases
 
 
-def get_open_episode_case_from_occurrence(domain, occurrence_case_id):
+def get_open_active_dstb_episode_case_from_occurrence(domain, occurrence_case_id):
     """
-    Gets the first open 'episode' case for the occurrence
+    In eNikshay public sector app
+    Gets the first open DSTB 'episode' case for the occurrence
 
     Assumes the following case structure:
     Occurrence <--ext-- Episode
 
     """
     case_accessor = CaseAccessors(domain)
     episode_cases = case_accessor.get_reverse_indexed_cases([occurrence_case_id])
-    open_episode_cases = [case for case in episode_cases
-                          if not case.closed and case.type == CASE_TYPE_EPISODE and
-                          case.dynamic_case_properties().get('episode_type') == "confirmed_tb"]
-    if open_episode_cases:
-        return open_episode_cases[0]
+    open_active_episode_cases = [case for case in episode_cases
+                                 if not case.closed and
+                                 case.type == CASE_TYPE_EPISODE and
+                                 case.get_case_property().get('episode_type') == "confirmed_tb" and
+                                 case.get_case_property('is_active') == 'true'
+                                 ]
+    if len(open_active_episode_cases) > 1:
+        raise EnikshayBadAppState("Multiple open active episode cases found for occurrence")
+
+    if open_active_episode_cases:
+        return open_active_episode_cases[0]
     else:
         raise ENikshayCaseNotFound(
             "Occurrence with id: {} exists but has no open episode cases".format(occurrence_case_id)
         )
 
 
-def get_open_drtb_hiv_case_from_episode(domain, episode_case_id):
+def get_open_drtb_hiv_case_from_occurrence(domain, occurrence_case_id):
```

Comment:
@sheelio this would now return a `secondary_owner` case. Earlier this was returning `CASE_TYPE_DRTB_HIV_REFERRAL`.

Reply:
That sounds right - where is it used?

Reply:
https://github.com/dimagi/commcare-hq/blob/e635e762a9d0e6d48a5695a571965d3835948e56/custom/enikshay/nikshay_datamigration/factory.py#L120

and then in `get_drtb_hiv_referral_case_structure` -> `get_case_structures_to_create`

Reply:
@sheelio please confirm this.

Reply:
I'm not sure how to best follow the code, but the the underlying change is that instead of DRTB HIV cases we have secondary owner cases.  As long as we update all the places that refer to getting a DRTB HIV case to using a secondary_owner case, that's fine.

Reply:
This is used for data migration. Usage [here](https://github.com/dimagi/commcare-hq/blob/master/custom/enikshay/nikshay_datamigration/factory.py#L114).
Hey @nickpell can you suggest what the corresponding change here would be for this update that we are doing in this PR for this method to return a `secondary_owner` case and not `drtb-hiv-referral` case.